pkg/build/controller/strategy/util.go

```diff
@@ -193,6 +164,25 @@ func addOriginVersionVar(output *[]kapi.EnvVar) {
 	*output = append(*output, version)
 }
 
+// addOutputEnvVars adds env variables that provide information about the output
+// target for the build
+func addOutputEnvVars(buildOutput *kapi.ObjectReference, output *[]kapi.EnvVar) error {
```

Comment:
no particular disagreement with that philosophy, it's definitely not how i'd normally implement it, but i was just staying in sync w/ how the rest of these methods(the other ones that add environment values) are already implemented rather than either having one be different, or refactoring all of them and all of the code which invokes it, which is a more significant change with a higher risk of breaking something.

Also given the name of the method is "addXXXX" there is a certain convenience+logic to the idea that you pass in the array you want things added to, rather than having to get a result back and then append that result to something.